cli/resolver.rs

```diff
@@ -33,3 +35,42 @@ impl Resolver for ImportMapResolver<'_> {
       .map_err(|err| err.into())
   }
 }
+
+#[derive(Debug, Default, Clone)]
+pub(crate) struct JsxResolver {
+  jsx_import_source_module: String,
+  maybe_import_map_resolver: Option<ImportMapResolver>,
```

Comment:
Just a thought and maybe not for this review, but I wonder if we should have some other `AggregateResolver` or something that takes a collection of `dyn Resolver`s and then tries to resolve with each resolver in order. That way in the consuming code someone could build one up and it might be a bit more clear about what's going on.

Reply:
See: #12684